packages/core/contracts-ovm/insured-bridge/implementation/OVM_BridgeDepositBox.sol

```diff
@@ -1,47 +1,237 @@
+
+    event L1WithdrawContractChanged(address oldL1WithdrawContract, address newL1WithdrawContract);
+    event TokenWhitelisted(address l1Token, address l2Token);
+    event DepositsEnabled(bool enabledResultantState);
+    event L2FundsDeposited(
         uint256 transferId,
         uint256 timestamp,
         address sender,
         address recipient,
-        address originToken,
+        address l1Token,
         uint256 amount,
-        uint256 networkId,
         uint256 maxFee
     );
-    event WithdrawalContractSet(address withdrawalContract);
-    event TokenWhitelisted(address originToken, address destinationToken);
-    event DepositContractEnabledStateToggled(bool enabled);
+    event TokensBridged(address l2Token, uint256 numberOfTokensBridged, uint256 l1Gas, address caller);
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
 
-    constructor(address _l2CrossDomainMessenger, address _l1Owner) OVM_CrossDomainEnabled(_l2CrossDomainMessenger) {
-        l1Owner = _l1Owner;
+    modifier onlyIfDepositsEnabled() {
+        require(depositsEnabled, "Contract is disabled");
+        _;
     }
 
-    // Admin functions
+    /**
+     * @notice Construct the OVM Bridge Deposit Box
+     * @param _l1WithdrawContract Address of the bridge withdraw contract on L1. Tokens are sent to this address at the
+     *          conclusion of fast relay. This address is the "owner" of the deposit box.
+     * @param _minimumBridgingDelay Minimum second that must elapse between L2->L1 token transfer to prevent dos.
+     * @param timerAddress Timer used to synchronize contract time in testing. Set to 0x000... in production.
+     */
+    constructor(
+        address _l1WithdrawContract,
+        uint64 _minimumBridgingDelay,
+        address timerAddress
+    ) OVM_CrossDomainEnabled(Lib_PredeployAddresses.L2_CROSS_DOMAIN_MESSENGER) OVM_Testable(timerAddress) {
+        l1WithdrawContract = _l1WithdrawContract;
+        minimumBridgingDelay = _minimumBridgingDelay;
+    }
 
-    function setWithdrawalContract(address withdrawalContract) public onlyFromCrossDomainAccount(l1Owner) {}
+    /**************************************
+     *          ADMIN FUNCTIONS           *
+     **************************************/
 
-    function whitelistToken(address l1Token, address l2Token) public onlyFromCrossDomainAccount(l1Owner) {}
+    /**
+     * @notice Changes the L1 withdraw associated with this L2 deposit box.
+     * @dev Only callable by the existing l1WithdrawContract via the optimism cross domain messenger.
+     * @param newL1WithdrawContract address of the new L1 withdrawContract.
+     */
+    function changeL1WithdrawContract(address newL1WithdrawContract)
+        public
+        onlyFromCrossDomainAccount(l1WithdrawContract)
+    {
+        emit L1WithdrawContractChanged(l1WithdrawContract, newL1WithdrawContract);
```

Comment:
nit: should emit the event after changing state

Reply:
I put it before intentionally. else, you'd need to store the `l1WithdrawContract` to get the `oldL1WithdrawContract` address.

Reply:
I think actually you should change the event signature to only emit the new withdrawal address. Then, in the constructor call an internal method `_setWithdrawContract(address newWithdrawContract)` which will emit the address set upon construction. This follows the pattern I use in `BridgeRouter`. 

On a side note, I don't actually set the Deposit contract in the constructor. WDYT about setting Deposit/Withdraw in constructor? My reasoning was depending on the order of deployments we may not know the addresses ahead of time

Reply:
sure, can do the event change and call the `setWithdrawContract` in the constructor to get the event.

On your side note, I do set the `l1WithdrawContract` contract in the constructor. However, we cant have _both_ the deposit and withdrawal contract set them in the constructor as you need to deploy one contract first, then the other. ideally, we deploy on L1 first, then deploy on L2 with the withdraw contract as a param in the deposit contract constructor then finally initialize the L1 contract with the L2 contract address.

Reply:
as a side note, I also need to add a function, event and tests for setting the `minimumBridgingDelay`. I will follow the same pattern where this is called in the constructor.